FreeRTOS_TCP_WIN.c

```diff
@@ -443,6 +443,14 @@
                 {
                     /* Could call vListInitialiseItem here but all data has been
                     * nulled already.  Set the owner to a segment descriptor. */
+
+                    #if ( configUSE_LIST_DATA_INTEGRITY_CHECK_BYTES == 1 )
```

Comment:
I'm happy with the change.

Reply:
The instances of list-items are properly initialised, like in [BufferAllocation_1](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/6d7f1a98b9c26fa7f4f7b884a8954cac348c9006/portable/BufferManagement/BufferAllocation_1.c#L207), in [BufferAllocation_2](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/6d7f1a98b9c26fa7f4f7b884a8954cac348c9006/portable/BufferManagement/BufferAllocation_2.c#L147), and in [FreeRTOS_Sockets.c](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/6d7f1a98b9c26fa7f4f7b884a8954cac348c9006/FreeRTOS_Sockets.c#L448)

This PR will add initialisation in [FreeRTOS_TCP_WIN.c](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/6d7f1a98b9c26fa7f4f7b884a8954cac348c9006/FreeRTOS_TCP_WIN.c#L446).

One more change is needed, in FreeRTOS_DNS.c, around line 390:
~~~
     pxCallback->uxRemaningTime = uxTimeout;
     vTaskSetTimeOutState( &pxCallback->uxTimeoutState );
+    vListInitialiseItem( &( pxCallback->xListItem ) );
     listSET_LIST_ITEM_OWNER( &( pxCallback->xListItem ), ( void * ) pxCallback );
~~~
@AniruddhaKanhere , can you take care of that?

Also I think that in FreeRTOS_TCP_WIN.c, we can call `vListInitialiseItem()` unconditionally:

~~~c
-    #if ( configUSE_LIST_DATA_INTEGRITY_CHECK_BYTES == 1 )
-    {
         vListInitialiseItem( &( xTCPSegments[ xIndex ].xSegmentItem ) );
         vListInitialiseItem( &( xTCPSegments[ xIndex ].xQueueItem ) );
-    }
-    #endif
~~~
It will not slow-down the application because this is only called at start-up time. And I noticed that in other instances `vListInitialiseItem()` is called unconditionally.

Reply:
I'll add that in a separate PR. Thanks @htibosch.